sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/FilterEstimation.scala

```diff
@@ -427,7 +427,11 @@ case class FilterEstimation(plan: Filter) extends Logging {
 
     // return the filter selectivity.  Without advanced statistics such as histograms,
     // we have to assume uniform distribution.
-    Some(math.min(newNdv.toDouble / ndv.toDouble, 1.0))
+    if (ndv.toDouble != 0) {
```

Comment:
What's the concrete case when `ndv.toDouble == 0`?
Also, is this only an place where we need this check? 
For example, we don't here:
https://github.com/apache/spark/blob/5cfd5fabcdbd77a806b98a6dd59b02772d2f6dee/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/EstimationUtils.scala#L166

Reply:
I have experienced this problem for the sub condition with IN clause, smth like  "FLD in ("value")".
To my mind,  this happens, if the table is empty. In this case ndv will be 0. 
I think, it will make sense, to check it everywhere it is used in this way.

Reply:
Can you add a test for the empty table case?
I think we need to fix the other places if they have the same issue. cc: @wzhfy

Reply:
Can you add a test for the empty table case?
I think we need to fix the other places if they have the same issue. cc: @wzhfy

Reply:
I have developed the test that illustrates the problem. It is just a small scala app right now.
Right now it breaks after the last select. Adding `REFRESH TABLE TBL`  does not fix the problem in this particular case. 
Should I add it to some particular test suite ? 

        object SparkCBOStatisticsTest {
          def main(args: Array[String]): Unit = {
            val spark = SparkSession.builder
              .master("local[*]")
              .config("spark.sql.adaptive.enabled", "true")
              .config("spark.sql.cbo.enabled", "true")
              .config("spark.sql.cbo.joinReorder.enabled", "true")
              .config("spark.sql.sources.bucketing.enabled", "true")
              .getOrCreate()
            import spark.implicits._
            import org.apache.spark.sql.functions._
            val df = spark.range(10000L).select('id,
              'id * 2 as "fld1",
              'id * 12 as "fld2",
              lit("aaa") + 'id as "fld3",
              lit("bbb") + 'id * 'id as "fld4")
            df.write
              .mode(SaveMode.Overwrite)
              .bucketBy(10, "id","fld1", "fld2")
              .sortBy("id","fld1", "fld2")
              .saveAsTable("TBL")
            spark.sql("ANALYZE TABLE TBL COMPUTE STATISTICS ")
            spark.sql("ANALYZE TABLE TBL COMPUTE STATISTICS FOR COLUMNS ID, FLD1, FLD2, FLD3")
            val df2 = spark.sql("select t1.fld1*t2.fld2 as fld1, t1.fld2, t1.fld3 from tbl t1 join tbl t2 on t1.id=t2.id where t1.fld1>100 and t1.fld3 in (-123.23,321.23) ")
            df2.explain(true)
            df2.show(100)
            df2.createTempView("tbl2")
            spark.sql("DESC FORMATTED TBL").show(false)
            spark.sql("select * from tbl2 where fld3 in ('qqq', 'qwe') and fld1<100  ").show(100, false)
        
          }
        }